src/provisioner/provisioner.py

```diff
@@ -218,18 +195,20 @@ def _get_node_info(cortx_conf: ConfigStore):
     def _provision_components(cortx_conf: ConfigStore, mini_prov_interfaces: list, apply_phase: str):
         """Invoke Mini Provisioners of cluster components."""
         node_id, node_name = CortxProvisioner._get_node_info(cortx_conf)
-        components = cortx_conf.get(f'node>{node_id}>components')
-        if components is None:
-            Log.warn(f"No component specified for {node_name} in CORTX config")
-        num_components = len(components)
+        num_components = int(cortx_conf.get(f'node>{node_id}>num_components'))
         for interface in mini_prov_interfaces:
             for comp_idx in range(0, num_components):
-                services = cortx_conf.get(
-                    f'node>{node_id}>components[{comp_idx}]>services')
-                service = 'all' if services is None else ','.join(services)
-                component_name = components[comp_idx]['name']
+                key_prefix = f'node>{node_id}>components[{comp_idx}]'
+                component_name = cortx_conf.get(f'{key_prefix}>name')
+                # Get services.
+                service_idx = 0
+                services = []
+                while (cortx_conf.get(f'{key_prefix}>services[{service_idx}]') is not None):
```

Comment:
This is not the absolute key. We are accessing non-leaf key.

Reply:
Here we are accessing the leaf key only.
`node:
  8efd697708a8f7e428d3fd520c180795:
    - name: s3
      services:
      - io
      - auth
      - bg_consumer
`
Also tested this change with consul, its working.